test/basic_app/main.c

```diff
@@ -3,23 +3,38 @@ volatile int test = 1;
 // bin
 const char bin_data[] = {0,1,2,3,4,5,6,7};
```

Comment:
Probably worth a comment on each variable to specify which section they end up in (e.g. rodata for all the const ones, data for `switch_arg` and `test`).

Reply:
The obdump output doesn't seem to tell what the section types are though? https://gist.github.com/sozud/229676a04a3a371d4f1e51d8fb0986b4
I ran `mips-linux-gnu-objdump -D -z build/basic_app.elf > basic_app.txt`

Reply:
You can't rely on symbol data in the linked .elf to know which section a variable is in since they'll all be in `.boot` by that point due to the linker script. If you run `mips-linux-gnu-readelf -Ss build/basic_app.elf` you'll see that the symbols are mostly in section indices 3 and 4, which correspond to `.boot` and `.boot_bss`. You can figure out what section the symbols originally came from by looking at their addresses relative to the section start/end symbols, looking at the map file, or using readelf/objdump on the unlinked .o files.

Reply:
Thanks for the explanation. If I dump the map file I get this: https://gist.github.com/sozud/9e8f74d030ced4dd4ecced720b8a6499
Is bss_data still bss despite being marked COMMON?
```
 build/main.o(COMMON)
 COMMON         0x0000000080000540       0x80 build/main.o
                0x0000000080000540                bss_data
```

Reply:
COMMON becomes bss during linking, it's a way for multiple object files to define a symbol without causing multiple definitions errors.

Reply:
Thanks, all the variables are marked now.